.github/actions/update-charm-pins/main.py

```diff
@@ -49,5 +49,7 @@ def update_charm_pins(workflow):
 
 if __name__ == '__main__':
     logging.basicConfig(level='INFO')
-    for workflow in ' '.join(sys.argv[1:]).split():
+    workflows = os.environ['WORKFLOWS'].split()
+    for workflow in workflows:
+        assert re.match(r'^[a-zA-Z\-]+.yaml$', workflow), 'Use a safer name.'
```

Comment:
If we're dropping command line arguments:
- please update the sibling `readme.md`
- issue an error or warning if the env var is empty

The original ask was that this script could be used from command line in the local checkout. That's also half the reason that it's a composite action.

Reply:
The regexp does not look quite right... the inputs are like:
```
.github/workflows/db-charm-tests.yaml
.github/workflows/hello-charm-tests.yaml
.github/workflows/charmcraft-pack.yaml
.github/workflows/observability-charm-tests.yaml
```

Current version won't allow leading dot or the slashes in the middle.
At the same time, we probably want to prevent random PRs passing things like `/etc/netplan/01-network-manager-all.yaml` in...

I'm not sure what's better:
- no validation?
- better regexp?
- `Path(inp).absolute().is_relative_to(Path(".github").absolute())` ?

Reply:
What about just going back to passing the values as a command line argument? It seems like that would be much simpler.

Reply:
I'm not against that!